## 시작

In [1]:
!pip install --upgrade pip
!python -m pip install --user --upgrade pip

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 46.8 MB/s eta 0:00:00
  Attempting uninstall: pip
    Found existing installation: pip 24.1.2
    Uninstalling pip-24.1.2:
      Successfully uninstalled pip-24.1.2


In [2]:
# 전처리용 모듈 설치(모델링용 설치는 Applying Algorithms 뒤에서 하겠습니다.)
import warnings
warnings.filterwarnings('ignore') # 경고 메시지 무시

# System & Time
import os

# Data Manipulation
import pandas as pd
import numpy as np
from tqdm import tqdm # 진행상황 표시

# Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Statistics & Scaling (전처리에 필요한 통계 및 스케일링)
import statsmodels.api as sm
from scipy import stats
from statsmodels.stats.outliers_influence import variance_inflation_factor # VIF 확인용
from sklearn.preprocessing import MinMaxScaler, StandardScaler # 스케일링
from sklearn import preprocessing


# Pandas 출력 옵션 설정
pd.options.display.float_format = '{:,.2f}'.format
pd.options.display.max_rows = 150
pd.options.display.max_columns = None

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## 1. 사업과제정보

### 초기분석단계(별 성과 없음. 열람 시 바쁘신 분들은 넘기셔도 됨)


In [4]:
# 1. df_사업과제 데이터 할당 및 확인
df_사업과제 = pd.read_csv('/content/drive/MyDrive/DIVE_실습/brand diving/data/r&d 예산배분 최적화 논문 데이터(결합)/사업과제정보.csv')

print(df_사업과제.columns.tolist())
display(df_사업과제.head(20))

['과제수행연도', '사업명', '사업구분', '부처명', '신규계속구분코드', '신규계속구분', '이전(기관)연구개발과제번호', '이전과제고유번호', '내역사업명', '(기관)연구개발과제번호', '과제고유번호', '총괄주관(기관)연구개발과제번호', '총괄주관과제고유번호', '연구개발과제명-국문', '총연구기간-시작연월일', '총연구기간-종료연월일', '총괄주관/주관구분', '연구수행주체코드', '연구수행주체', '지역코드', '지역', '기초자치단체코드', '기초자치단체명', '과학기술표준분류코드1-대', '과학기술표준분류1-대', '과학기술표준분류코드1-중', '과학기술표준분류1-중', '과학기술표준분류코드1-소', '과학기술표준분류1-소', '과학기술표준분류가중치1', '과학기술표준분류코드2-대', '과학기술표준분류2-대', '과학기술표준분류코드2-중', '과학기술표준분류2-중', '과학기술표준분류코드2-소', '과학기술표준분류2-소', '과학기술표준분류가중치2', '과학기술표준분류코드3-대', '과학기술표준분류3-대', '과학기술표준분류코드3-중', '과학기술표준분류3-중', '과학기술표준분류코드3-소', '과학기술표준분류3-소', '과학기술표준분류가중치3', '중점과학기술분류코드-대', '중점과학기술분류-대', '중점과학기술분류코드-중', '중점과학기술분류-중', '중점과학기술분류코드-소', '중점과학기술분류-소', '국가전략기술코드-전략기술', '국가전략기술-전략기술', '국가전략기술코드-세부중점기술', '국가전략기술-세부중점기술', '적용분야코드1', '적용분야1', '적용분야가중치1', '적용분야코드2', '적용분야2', '적용분야가중치2', '적용분야코드3', '적용분야3', '적용분야가중치3', '녹색기술분류코드', '녹색기술분류', '6T관련기술코드-대', '6T관련기술-대', '6T관련기술코드-중', '6T관련기술-중', '6T관련기술코드-소', '6T관련기술-소', '경제사회목적코드', '경제사회목적', '(2014~2017)국가전략기술

,과제수행연도,사업명,사업구분,부처명,신규계속구분코드,신규계속구분,이전(기관)연구개발과제번호,이전과제고유번호,내역사업명,(기관)연구개발과제번호,과제고유번호,총괄주관(기관)연구개발과제번호,총괄주관과제고유번호,연구개발과제명-국문,총연구기간-시작연월일,총연구기간-종료연월일,총괄주관/주관구분,연구수행주체코드,연구수행주체,지역코드,지역,기초자치단체코드,기초자치단체명,과학기술표준분류코드1-대,과학기술표준분류1-대,과학기술표준분류코드1-중,과학기술표준분류1-중,과학기술표준분류코드1-소,과학기술표준분류1-소,과학기술표준분류가중치1,과학기술표준분류코드2-대,과학기술표준분류2-대,과학기술표준분류코드2-중,과학기술표준분류2-중,과학기술표준분류코드2-소,과학기술표준분류2-소,과학기술표준분류가중치2,과학기술표준분류코드3-대,과학기술표준분류3-대,과학기술표준분류코드3-중,과학기술표준분류3-중,과학기술표준분류코드3-소,과학기술표준분류3-소,과학기술표준분류가중치3,중점과학기술분류코드-대,중점과학기술분류-대,중점과학기술분류코드-중,중점과학기술분류-중,중점과학기술분류코드-소,중점과학기술분류-소,국가전략기술코드-전략기술,국가전략기술-전략기술,국가전략기술코드-세부중점기술,국가전략기술-세부중점기술,적용분야코드1,적용분야1,적용분야가중치1,적용분야코드2,적용분야2,적용분야가중치2,적용분야코드3,적용분야3,적용분야가중치3,녹색기술분류코드,녹색기술분류,6T관련기술코드-대,6T관련기술-대,6T관련기술코드-중,6T관련기술-중,6T관련기술코드-소,6T관련기술-소,경제사회목적코드,경제사회목적,(2014~2017)국가전략기술코드-대,(2014~2017)국가전략기술-대,(2014~2017)국가전략기술코드-중,(2014~2017)국가전략기술-중,(2014~2017)국가전략기술코드-소,(2014~2017)국가전략기술-소,요약문_연구목표,요약문_연구내용,요약문_기대효과,요약문_한글키워드,요약문_영문키워드,당해연구기간-시작연월일,당해연구기간-종료연월일,대표전문기관,과제관리기관,과제수행기관명(정비),과제수행기관사업자등록번호(정비),연구개발단계코드(변경),연구개발단계(변경),세부과제지원유형코드,세부과제지원유형명,정부지원연구개발비(원),인건비_현금,인건비_현물,직접비_현금,직접비_현물,간접비,위탁연구개발비(원),물건비,민간연구비_지방정부_현금,민간연구비_지방정부_현물,민간연구비_대학_현금,민간연구비_대학_현물,민간연구비_대기업_현금,민간연구비_대기업_현물,민간연구비_중견기업_현금,민간연구비_중견기업_현물,민간연구비_중소기업_현금,민간연구비_중소기업_현물,민간연구비_기업_현금,민간연구비_기업_현물,민간연구비_기타_현금,민간연구비_기타_현물,민간연구비합계(원),총연구비_현금,총연구비_현물,총연구비_합계(원),공동연구여부_기업,공동연구여부_대학,공동연구여부_출연연,공동연구여부_외국,공동연구여부_기타,참여연구원_성별_남,참여연구원_성별_여,참여연구원_학위_박사,참여연구원_학위_석사,참여연구원_학위_학사이하,참여연구원_전공_이학,참여연구원_전공_공학,참여연구원_전공_농학,참여연구원_전공_의학,참여연구원_전공_인문,참여연구원_전공_기타,연구책임자소속기관명(정비),연구책임자사업자번호(정비),연구책임자성별,연구책임자전공,연구책임자학위,출처_파일명
0,2014,과학기술국제부담금(비ODA),일반연구개발사업,미래창조과학부,2,계속,2013K1A3A2A06056466,"1,711,009,645.00",OECD 글로벌사이언스포럼(GSF) 부담금,2013K1A3A2A06056466,1711022506,2013K1A3A2A06056466,"1,711,022,506.00",2013OECD 글로벌사이언스포럼(GSF) 부담금,2013-01-01,2020-12-31,주관,2,출연연구소,17,해외,NaN,NaN,OC,과학기술과 인문사회,OC99,기타 과학기술과 인문사회,NaN,NaN,100.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,13,기타,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,NaN,NaN,NaN,NaN,NaN,NaN,NaN,한국연구재단,한국연구재단,미래창조과학부,*******93*,4,기타,NaN,NaN,23000000,0,0,23000000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,23000000,0,23000000,N,N,N,N,N,1.00,0.00,0.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,한국연구재단,*******32*,남자,인문사회학,박사,2014년사업과제정보.xlsx
1,2014,과학기술국제부담금(비ODA),일반연구개발사업,미래창조과학부,2,계속,2013K1A3A2A05056464,"1,711,009,644.00","국제기구단체(HFSP, GBIF) 참여부담금",2013K1A3A2A05056464,1711022507,2013K1A3A2A05056464,"1,711,022,507.00","국제기구단체(HFSP, GBIF) 참여부담금",2013-01-01,2020-12-31,주관,2,출연연구소,17,해외,NaN,NaN,OC,과학기술과 인문사회,OC99,기타 과학기술과 인문사회,NaN,NaN,100.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,13,기타,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,NaN,NaN,NaN,NaN,NaN,NaN,NaN,한국연구재단,한국연구재단,미래창조과학부,*******93*,4,기타,NaN,NaN,1069000000,0,0,1069000000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1069000000,0,1069000000,N,N,N,N,N,1.00,0.00,0.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,한국연구재단,*******32*,남자,인문사회학,박사,2014년사업과제정보.xlsx
2,2014,과학기술국제부담금(비ODA),일반연구개발사업,미래창조과학부,2,계속,2013K1A3A2A06056465,"1,711,009,646.00",유럽핵입자물리연구소(CERN) 참여 부담금,2013K1A3A2A06056465,1711022505,2013K1A3A2A06056465,"1,711,022,505.00",2014유럽핵입자물리연구소(CERN) 참여 부담금,2013-01-01,2018-12-31,주관,2,출연연구소,17,해외,NaN,NaN,NB,물리학,NB01,입자/장물리,NaN,NaN,100.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X01,지식의 진보(비목적 연구),100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"70,000.00",위의 미래유망신기술(6T) 

In [ ]:
# 1. 2013년 ~ 2016년 데이터만 추출 (논문과 동일한 조건.. 이긴 한데 막상 분석에는 2014만 썻다는 걸 이 셀을 만들 당시에는 몰랐다)

df_paper = df_사업과제[(df_사업과제['과제수행연도'] >= 2014) & (df_사업과제['과제수행연도'] <= 2017)]

print(f"2013~2016년 과제 수: {len(df_paper)}건")
# 논문에서는 최종 필터링 후 458건의 대학 데이터를 썼다고 나옵니다.

2013~2016년 과제 수: 224033건


In [ ]:
# 2. 연구수행주체(기업, 대학, 출연연 등) 분포 확인
print("=== 연구수행주체 분포 ===")
print(df_paper['연구수행주체'].value_counts(dropna=False))

# 만약 여기서 '대학' 데이터만 남기고 싶다면? (논문의 샘플링 방법론 복제)
df_univ = df_paper[df_paper['연구수행주체'].str.contains('대학', na=False)]
print(f"대학 필터링 후 과제 수: {len(df_univ)}건")

=== 연구수행주체 분포 ===
연구수행주체
대학        86413
중소기업      33278
출연연구소     19770
국공립연구소    12331
기타         6935
중견기업       2324
대기업        1458
정부부처        244
Name: count, dtype: int64
대학 필터링 후 과제 수: 86413건


In [ ]:
# 3. 국가과학기술표준분류코드 Top 10 뽑아보기
print("=== 과학기술표준분류 Top 10 ===")
print(df_univ['과학기술표준분류코드1-대'].value_counts().head(10))

# 만약 이 결과에 EA02, NA09, EE11이 상위권에 뜬다면?
# -> 빙고! 저자가 이 컬럼을 빈도수 기준으로 내림차순 정렬한 뒤,
# 상위 3개만 남기고 나머지를 'Others'로 묶었다는 완벽한 증거입니다.

=== 과학기술표준분류 Top 10 ===
과학기술표준분류코드1-대
LC    15117
LA     8368
LB     8183
EE     5152
EA     4368
ED     3504
EI     2995
NC     2928
NB     2540
EB     2517
Name: count, dtype: int64


In [ ]:
# 4. 당해연도 연구비(단위: 원) 요약 통계
print("=== 총연구비 합계 통계 ===")
print(df_univ['총연구비_합계(원)'].describe())
# (주의: 논문은 달러(M) 단위로 환산했으므로 환율/단위 변환 고려 필요  )

=== 총연구비 합계 통계 ===
count            86,413.00
mean        162,167,871.16
std         861,122,170.84
min                   1.00
25%          40,000,000.00
50%          51,090,000.00
75%         112,500,000.00
max     120,000,000,000.00
Name: 총연구비_합계(원), dtype: float64


In [ ]:
df_사업과제.shape

(714816, 142)

In [ ]:
df_사업과제.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 714816 entries, 0 to 714815
Columns: 142 entries, 과제수행연도 to 출처_파일명
dtypes: float64(31), int64(34), object(77)
memory usage: 774.4+ MB


In [ ]:
df_사업과제.describe()

,과제수행연도,신규계속구분코드,과제고유번호,총괄주관과제고유번호,연구수행주체코드,지역코드,기초자치단체코드,과학기술표준분류가중치1,과학기술표준분류가중치2,과학기술표준분류가중치3,중점과학기술분류코드-대,중점과학기술분류코드-중,중점과학기술분류코드-소,국가전략기술코드-전략기술,국가전략기술코드-세부중점기술,적용분야가중치1,적용분야가중치2,적용분야가중치3,녹색기술분류코드,6T관련기술코드-대,6T관련기술코드-중,6T관련기술코드-소,경제사회목적코드,(2014~2017)국가전략기술코드-대,(2014~2017)국가전략기술코드-중,(2014~2017)국가전략기술코드-소,연구개발단계코드(변경),세부과제지원유형코드,정부지원연구개발비(원),인건비_현금,인건비_현물,직접비_현금,직접비_현물,간접비,위탁연구개발비(원),물건비,민간연구비_지방정부_현금,민간연구비_지방정부_현물,민간연구비_대학_현금,민간연구비_대학_현물,민간연구비_대기업_현금,민간연구비_대기업_현물,민간연구비_중견기업_현금,민간연구비_중견기업_현물,민간연구비_중소기업_현금,민간연구비_중소기업_현물,민간연구비_기업_현금,민간연구비_기업_현물,민간연구비_기타_현금,민간연구비_기타_현물,민간연구비합계(원),총연구비_현금,총연구비_현물,총연구비_합계(원),참여연구원_성별_남,참여연구원_성별_여,참여연구원_학위_박사,참여연구원_학위_석사,참여연구원_학위_학사이하,참여연구원_전공_이학,참여연구원_전공_공학,참여연구원_전공_농학,참여연구원_전공_의학,참여연구원_전공_인문,참여연구원_전공_기타
count,"714,816.00","714,816.00","714,816.00","697,011.00","714,816.00","714,816.00","552,063.00","633,582.00","309,845.00","211,754.00","331,426.00","331,426.00","331,426.00","130,125.00","130,125.00","714,816.00","215,989.00","132,972.00","307,807.00","653,371.00","653,371.00","653,371.00","714,816.00","194,654.00","194,654.00","194,654.00","714,816.00","490,783.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","699,263.00","699,239.00","699,192.00","699,134.00","699,149.00","699,123.00","699,236.00","698,982.00","698,999.00","699,088.00","699,175.00"
mean,"2,019.29",1.56,"1,609,406,761.34","1,611,987,755.81",7.24,7.91,"8,178.01",81.96,27.08,14.36,"8,461.19","84,613.57","84,613.57","149,102,480.15","149,102,592.55",91.12,23.91,8.91,902.36,"34,571.80","34,738.24","34,749.25",7.30,"34,242.09","34,417.23","34,419.84",2.10,124.49,"351,343,636.97","75,897,358.69","16,041,057.94","230,840,579.94","16,765,549.64","23,564,713.75","5,647,594.67","1,032,252.95","7,131,091.80","607,565.23","1,046,808.06","2,678,440.61","2,044,103.01","3,202,659.50","1,269,834.37","3,068,159.66","6,063,658.87","19,984,571.42","9,377,596.25","26,255,390.57","2,866,671.49","3,245,843.43","53,209,407.44","371,765,804.57","32,787,239.84","404,654,334.80",6.88,2.24,2.82,1.91,4.39,1.19,4.12,0.36,0.51,0.63,2.30
std,3.06,0.50,"604,064,004.24","618,374,910.69",18.92,13.12,"13,276.42",21.84,13.20,8.73,"2,962.17","29,621.49","29,621.49","40,031.73","39,884.15",17.35,16.74,9.29,241.50,"22,500.38","22,415.02","22,410.07",3.02,"18,888.52","18,785.79","18,784.34",1.11,41.95,"3,814,219,121.13","952,385,289.98","86,078,491.25","3,225,065,814.78","922,310,820.12","259,104,760.90","43,715,926.78","236,143,163.56","179,491,680.17","47,861,171.10","35,113,774.21","47,833,191.58","88,597,199.99","66,716,301.74","49,132,282.73","50,328,511.57","60,508,760.44","774,675,058.73","121,360,664.08","781,827,515.24","601,413,303.86","488,229,811.27","1,149,915,374.16","3,886,509,089.46","928,474,445.68","4,012,915,876.10",18.32,5.81,11.46,6.73,10.61,5.35,15.78,1.92,2.94,3.73,8.35
min,"2,014.00",1.00,"1,055,000,476.00",0.00,1.00,1.00,"1,010.00",9.00,0.00,0.00,"1,010.00","10,101.00","10,101.00","149,010,000.00","149,010,100.00",35.00,0.00,0.00,111.00,"10,000.00","10,100.00","10,111.00",1.00,"10,000.00","10,100.00","10,101.00",1.00,101.00,0.00,"-30,767,000.00",0.00,"-59,860,000.00","-1,006,756,000.00","-2,834,000.00",0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,"-15,000.00",0.00,"-15,000.00",0.00,"-100,000,000.00",0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,-1.00
25%,"2,017.00",1.00,"1,395,035,018.75","1,395,047,123.00",3.00,1.00,"1,085.00",60.00,20.00,10.00,"7,010.00","70,102.00","70,102.00","149,070,000.00","149,070,200.00",100.00,10.00,0.00,999.00,"20,000.00","20,100.00","20,115.00",5.00,"20,000.00","20,100.00","20,111.00",1.00,101.00,"41,766,750.00",0.00,0.00,"16,287,000.00",0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,

In [ ]:
df_사업과제.describe(include='all')

,과제수행연도,사업명,사업구분,부처명,신규계속구분코드,신규계속구분,이전(기관)연구개발과제번호,이전과제고유번호,내역사업명,(기관)연구개발과제번호,과제고유번호,총괄주관(기관)연구개발과제번호,총괄주관과제고유번호,연구개발과제명-국문,총연구기간-시작연월일,총연구기간-종료연월일,총괄주관/주관구분,연구수행주체코드,연구수행주체,지역코드,지역,기초자치단체코드,기초자치단체명,과학기술표준분류코드1-대,과학기술표준분류1-대,과학기술표준분류코드1-중,과학기술표준분류1-중,과학기술표준분류코드1-소,과학기술표준분류1-소,과학기술표준분류가중치1,과학기술표준분류코드2-대,과학기술표준분류2-대,과학기술표준분류코드2-중,과학기술표준분류2-중,과학기술표준분류코드2-소,과학기술표준분류2-소,과학기술표준분류가중치2,과학기술표준분류코드3-대,과학기술표준분류3-대,과학기술표준분류코드3-중,과학기술표준분류3-중,과학기술표준분류코드3-소,과학기술표준분류3-소,과학기술표준분류가중치3,중점과학기술분류코드-대,중점과학기술분류-대,중점과학기술분류코드-중,중점과학기술분류-중,중점과학기술분류코드-소,중점과학기술분류-소,국가전략기술코드-전략기술,국가전략기술-전략기술,국가전략기술코드-세부중점기술,국가전략기술-세부중점기술,적용분야코드1,적용분야1,적용분야가중치1,적용분야코드2,적용분야2,적용분야가중치2,적용분야코드3,적용분야3,적용분야가중치3,녹색기술분류코드,녹색기술분류,6T관련기술코드-대,6T관련기술-대,6T관련기술코드-중,6T관련기술-중,6T관련기술코드-소,6T관련기술-소,경제사회목적코드,경제사회목적,(2014~2017)국가전략기술코드-대,(2014~2017)국가전략기술-대,(2014~2017)국가전략기술코드-중,(2014~2017)국가전략기술-중,(2014~2017)국가전략기술코드-소,(2014~2017)국가전략기술-소,요약문_연구목표,요약문_연구내용,요약문_기대효과,요약문_한글키워드,요약문_영문키워드,당해연구기간-시작연월일,당해연구기간-종료연월일,대표전문기관,과제관리기관,과제수행기관명(정비),과제수행기관사업자등록번호(정비),연구개발단계코드(변경),연구개발단계(변경),세부과제지원유형코드,세부과제지원유형명,정부지원연구개발비(원),인건비_현금,인건비_현물,직접비_현금,직접비_현물,간접비,위탁연구개발비(원),물건비,민간연구비_지방정부_현금,민간연구비_지방정부_현물,민간연구비_대학_현금,민간연구비_대학_현물,민간연구비_대기업_현금,민간연구비_대기업_현물,민간연구비_중견기업_현금,민간연구비_중견기업_현물,민간연구비_중소기업_현금,민간연구비_중소기업_현물,민간연구비_기업_현금,민간연구비_기업_현물,민간연구비_기타_현금,민간연구비_기타_현물,민간연구비합계(원),총연구비_현금,총연구비_현물,총연구비_합계(원),공동연구여부_기업,공동연구여부_대학,공동연구여부_출연연,공동연구여부_외국,공동연구여부_기타,참여연구원_성별_남,참여연구원_성별_여,참여연구원_학위_박사,참여연구원_학위_석사,참여연구원_학위_학사이하,참여연구원_전공_이학,참여연구원_전공_공학,참여연구원_전공_농학,참여연구원_전공_의학,참여연구원_전공_인문,참여연구원_전공_기타,연구책임자소속기관명(정비),연구책임자사업자번호(정비),연구책임자성별,연구책임자전공,연구책임자학위,출처_파일명
count,"714,816.00",714816,714816,714816,"714,816.00",714816,398467,403225,714815,714815,"714,816.00",674915,"697,011.00",714816,714816,714816,714816,"714,816.00",714816,"714,816.00",714816,"552,063.00",552063,622723,633582,633582,633582,540712,540712,"633,582.00",269254,275331,275331,275331,248524,248524,"309,845.00",163543,166437,166437,166437,156134,156134,"211,754.00","331,426.00",331426,"331,426.00",331426,"331,426.00",331426,"130,125.00",130125,"130,125.00",130125,714816,714816,"714,816.00",163852,163852,"215,989.00",70550,70550,"132,972.00","307,807.00",307807,"653,371.00",653371,"653,371.00",653371,"653,371.00",653371,"714,816.00",714816,"194,654.00",194654,"194,654.00",194654,"194,654.00",194654,650835,650843,650712,672148,642269,606890,606890,714816,714816,714816,714816,"714,816.00",714816,"490,783.00",490783,"714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00","714,816.00",644984,644984,644984,641310,644984,"699,263.00","699,239.00","699,192.00","699,134.00","699,149.00","699,123.00","699,236.00","698,982.00","698,999.00","699,088.00","699,175.00",714587,714816,664585,655962,656292,714816
unique,NaN,3944,4,47,NaN,2,246223,398558,14835,440753,NaN,349559,NaN,348884,3315,3625,2,NaN,8,NaN,19,NaN,208,36,39,416,503,2827,2634,NaN,35,38,402,489,2718,2568,NaN,35,38,401,489,2624,2497,NaN,NaN,12,NaN,44,NaN,121,NaN,13,NaN,52,33,33,NaN,33,33,NaN,33,33,NaN,NaN,59,NaN,7,NaN,23,NaN,104,NaN,13,NaN,6,NaN,21,NaN,121,452256,497992,455562,352211,328212,2413,2710,137,323,60606,103,NaN,4,NaN,3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2,2,2,2,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,57420,104,2,6,3,11
top,NaN,개인기초연구(과기정통부)(R&D),일반연구개발사업,교육부,NaN,계속,202006152,1711149676,기본연구,정책연구개발-01,NaN,0,NaN,보안과제정보,2017-06-01,2023-12-31,주관,NaN,대학,NaN,서울특별시,NaN,유성구,LC,보건의료,LC02,임상의학,EE0202,S/W 솔루션,NaN,LC,보건의료,LC01,의생명과학,EE0108,인공지능,NaN,LC,보건의료,LC01,의생명과학,EE0108,인공지능,NaN,NaN,생명·보건의료,NaN,기타,NaN,위의 중점과학기술에 속하지 않는 기타 연구,NaN,위의 국가전략기술 분류에 속하지 않는 기타 연구,NaN,위의 국가전략기술 분류에 속하지 않는 기타 연구,X02,건강,NaN,X02,건강,NaN,X99,기타 공공목적,NaN,NaN,녹색기술관련 과제 아님,NaN,BT(생명공학기술),NaN,위의 미래유망신기술(6T) 103개 세분류에 속하지 않

In [ ]:
df_사업과제['과제고유번호'].value_counts(normalize=True) * 100

,proportion
과제고유번호,
2410000151,0.00
2410002542,0.00
2410000629,0.00
2410000626,0.00
2410000608,0.00
...,...
1711022508,0.00
1711022511,0.00
1711022505,0.00


In [ ]:
df_사업과제['사업명'].value_counts().head(100)

,count
사업명,
개인기초연구(과기정통부)(R&D),48018
개인기초연구(과기정통부),43315
인문사회기초연구,33809
이공학학술연구기반구축,23586
개인기초연구(교육부)(R&D),21743
인문사회기초연구(R&D),21239
이공학학술연구기반구축(R&D),16909
창업성장기술개발,16869
이공학개인기초연구지원,11320


In [ ]:
(df_사업과제.isnull().sum() / len(df_사업과제)) * 100

,0
과제수행연도,0.00
사업명,0.00
사업구분,0.00
부처명,0.00
신규계속구분코드,0.00
신규계속구분,0.00
이전(기관)연구개발과제번호,44.26
이전과제고유번호,43.59
내역사업명,0.00
(기관)연구개발과제번호,0.00


In [ ]:
df_사업과제.groupby('부처명')['정부지원연구개발비(원)'].sum()

,정부지원연구개발비(원)
부처명,
개인정보보호위원회,17196000000
경찰청,253107141801
고용노동부,6411622530
공정거래위원회,2542384634
과학기술정보통신부,62824081998506
관세청,279000000
교육부,21279432302301
국가유산청,58436616550
국무조정실,5576323694410


In [ ]:
df_사업과제.groupby('과제고유번호')['정부지원연구개발비(원)'].sum()

,정부지원연구개발비(원)
과제고유번호,
1055000476,1900000000
1055000477,350000000
1055000478,40000000
1055000479,49191000
1055000480,35000000
...,...
9991009052,270000000
9991009053,128000000
9991009054,297000000


In [ ]:
df_사업과제['부처명'].value_counts()

,count
부처명,
교육부,173165
과학기술정보통신부,170415
중소벤처기업부,85290
산업통상자원부,58786
농촌진흥청,44088
미래창조과학부,36546
보건복지부,26467
중소기업청,22708
농림축산식품부,17738


In [ ]:
len(df_사업과제[df_사업과제['정부지원연구개발비(원)'] >= 1000000000])

41300

### 선행연구의 변수들을 알아내자.

In [ ]:
unique_values = df_사업과제['연구수행주체'].unique()

In [ ]:
print(unique_values)

['출연연구소' '대학' '기타' '정부부처' '중소기업' '대기업' '중견기업' '국공립연구소']


In [ ]:
print(df_사업과제['연구책임자전공'].unique())

['인문사회학' '공학' '기타' '이학' '농림수산학' '의약보건학' nan]


In [ ]:
len(df_사업과제['연구책임자전공'].unique())

7

In [ ]:
len(df_사업과제['과학기술표준분류1-대'].unique())

40

In [ ]:
len(df_paper['과학기술표준분류1-대'].unique())

35

ㄴ 선행연구는 33개

In [ ]:
df_사업과제['과학기술표준분류코드1-대'].unique()

array(['OC', 'NB', 'EE', 'EA', 'NC', 'EG', 'LA', 'LC', 'ND', 'EH', 'EB',
       'EF', 'ED', 'EC', 'SH', 'LB', 'HE', 'SB', 'SI', 'SC', 'EI', 'SD',
       'OA', nan, 'OB', 'HB', 'HD', 'SF', 'SE', 'HC', 'ZZ', 'SA', 'SG',
       'HA', 'HG', 'HH', 'HF'], dtype=object)

In [ ]:
df_사업과제['과학기술표준분류1-대'].unique()

array(['과학기술과 인문사회', '물리학', '정보/통신', '기계', '화학', '원자력', '생명과학', '보건의료',
       '지구과학(지구/대기/해양/천문)', '환경', '재료', '에너지/자원', '전기/전자', '화공', '교육',
       '농림수산식품', '문화/예술/체육', '정치/행정', '미디어/커뮤니케이션/문헌정보', '경제/경영', '건설/교통',
       '사회/인류/복지/여성', '뇌과학', '수학', '인지/감성과학', '철학/종교', '문학', '지리/지역/관광',
       '생활', '언어', '기타', '법', nan, '심리', '역사/고고학', '사회과학', '과학기술과 인문사회 ',
       '문화예술체육학', '인문학', '인지･감성과학 '], dtype=object)

In [ ]:
df_사업과제['과학기술표준분류1-중'].unique()

array(['기타 과학기술과 인문사회', '입자/장물리', '소프트웨어', '인공위성', '광대역 통합망', '응집물질물리',
       '고분자화학', '기타 원자력', '분류/생태/환경생물학', '과학기술정책/사회', '원자핵물리', '보건학',
       '산업바이오', '의약품/의약품개발', '기타 생명과학', '면역학/생리학', '천문학', '융합화학',
       '발생/신경생물학', '원자로 계통/핵심기기 기술', '환경보건', '의생명과학', '기타 정보/통신',
       'ITS/텔레매틱스', '해양과학', '분자세포 생물학', '세라믹재료', '신재생에너지', '기타 재료', '기후학',
       '생화학/구조생물학', '융합바이오', '유전학/유전공학', '광화학', '정보통신 모듈/부품', '고분자재료',
       '위해성 평가/관리', '대기과학', '정보보호', '반도체소자/시스템', '물관리', '나노화학공정기술',
       '물리화학', '생물공학', '광학', '광응용기기', '생화학', '생물화학 공정기술', '유기화학', '홈네트워크',
       '나노화학', '기타 보건의료', '치료/진단기기', '방사선기술', '원자력기반/첨단기술', '조선/해양시스템',
       '기타 에너지/자원', '로봇/자동화기계', '우주과학', '열/표면처리', '금속재료', '항공시스템',
       '자원탐사/개발/활용', '교육일반', '의약품안전관리', '산림자원학', '농화학', '치의과학',
       '기타 전기/전자', '기타 교육', '무기센서 및 제어', '원자로 노심 기술', '제품디자인', '기타 정치/행정',
       '무기화학', '유체/플라즈마', '정보이론', 'U-컴퓨팅', '정보조직/검색/시스템', '위성/전파',
       '가정용기기/전자응용기기', '재무관리', '경영전략/윤리', 'RFID/USN', '계측기기', '전력IT',
       '이동통신', '에너지/환경

In [ ]:
len(df_사업과제['과학기술표준분류1-중'].unique())

504

In [ ]:
len(df_사업과제['과학기술표준분류1-소'].unique())

2635

In [ ]:
len(df_paper['과학기술표준분류1-중'].unique())

365

In [ ]:
len(df_paper['과학기술표준분류1-소'].unique())

2134

### 맞다 BK21 사업만 썼었지..! 이유도 있을 것이다 멍청하게 이걸 이제야 생각해내다니. 이걸 반영하여 선행연구들의 변수들을 알아내자


In [ ]:
target_col = '사업명'

# 1. 빈도수가 가장 높은 상위 5개의 값(Key) 추출
# value_counts()는 기본적으로 내림차순 정렬을 해주므로, head(5)로 맨 위 5개를 가져옵니다.
top_15_keys = df_사업과제[target_col].value_counts().head(15).index.tolist()

# [Hook] 추출된 상위 5개 값이 무엇인지 인간의 눈으로 직접 확인 (매우 중요!)
print(f"🔍 [{target_col}] 상위 5개 항목: {top_5_keys}")
print("-" * 50)

🔍 [사업명] 상위 5개 항목: ['개인기초연구(과기정통부)(R&D)', '개인기초연구(과기정통부)', '인문사회기초연구', '이공학학술연구기반구축', '개인기초연구(교육부)(R&D)', '인문사회기초연구(R&D)', '이공학학술연구기반구축(R&D)', '창업성장기술개발', '이공학개인기초연구지원', '중소기업기술혁신개발', '산학연협력기술개발', '창업성장기술개발(R&D)', '개인기초연구(교육부)', '개인기초연구(미래부)', '중소기업상용화기술개발']
--------------------------------------------------


때로는 원본 엑셀 전체파일 서칭도 필요하다. 2015년 사업과제정보 데이터에 BK21플러스사업 이라는 값의 사업명이 존재한다.

In [ ]:
# '컬럼명'에 '찾을값'이 입력된 행의 개수를 셉니다.
# (조건에 맞는 행은 True(1)가 되므로 이를 다 더하는 원리입니다)
target_count = (df_사업과제['사업명'] == 'BK21플러스사업').sum()

print(f"해당 샘플의 개수: {target_count:,}개")

해당 샘플의 개수: 1,336개


In [ ]:
df_bk21 = df_사업과제[df_사업과제['사업명'] == 'BK21플러스사업'].copy()

In [ ]:
# 1. 2013년 ~ 2016년 데이터만 추출 (논문과 동일한 조건)

df_bk21_1 = df_bk21[(df_bk21['과제수행연도'] >= 2014) & (df_bk21['과제수행연도'] <= 2017)]

print(f"2013~2016년 과제 수: {len(df_bk21_1)}건")

2013~2016년 과제 수: 1336건


이상한 일이다. BK21플러스사업이란 이름의 과제들이 모조리 2013~2016년이라고? 설마 전부 2015년인가? 다양한 버전의 BK21 사업 이름이 존재하는 건가


In [ ]:
df_bk21['과제수행연도'].unique()

array([2015])

역시나. BK21사업은 아마 해마다 다양한 버전의 이름으로 불렸을 것이다.


In [ ]:
# 기존에 작성하신 코드 (상위 15개 추출)
target_col = '사업명'
top_15_keys = df_사업과제[target_col].value_counts().head(25).index.tolist()

# [Hook] 리스트를 세로로 넘버링하여 출력
print(f"🔍 [{target_col}] 상위 15개 항목:")
print("-" * 40)
for i, key in enumerate(top_15_keys, 1):
    print(f"{i}. {key}")
print("-" * 40)

🔍 [사업명] 상위 15개 항목:
----------------------------------------
1. 개인기초연구(과기정통부)(R&D)
2. 개인기초연구(과기정통부)
3. 인문사회기초연구
4. 이공학학술연구기반구축
5. 개인기초연구(교육부)(R&D)
6. 인문사회기초연구(R&D)
7. 이공학학술연구기반구축(R&D)
8. 창업성장기술개발
9. 이공학개인기초연구지원
10. 중소기업기술혁신개발
11. 산학연협력기술개발
12. 창업성장기술개발(R&D)
13. 개인기초연구(교육부)
14. 개인기초연구(미래부)
15. 중소기업상용화기술개발
16. 일반연구자지원
17. 국방기술개발
18. 개인연구지원
19. 신진연구자지원
20. 소재부품기술개발
21. BK21플러스사업(R&D)
22. 중견연구자지원
23. 산학연협력기술개발(R&D)
24. 중소기업상용화기술개발(R&D)
25. 공정·품질기술개발(R&D)
----------------------------------------


In [6]:
target_col = '사업명'
ministry_col = '부처명'

# 찾고자 하는 키워드 및 부처명 입력
keyword = 'BK|두뇌|플러스'
target_ministry = '교육부'

# ---------------------------------------------------------
# [조건 1] 사업명에 키워드가 포함된 행 (기존)
mask_keyword = df_사업과제[target_col].str.contains(keyword, na=False, case=False)

# [조건 2] 부처명이 특정 부처와 정확히 일치하는 행
# (만약 부처명도 '포함'되는 것으로 찾고 싶다면 mask_keyword처럼 str.contains를 쓰셔도 됩니다)
mask_ministry = df_사업과제[ministry_col] == target_ministry

# 1. 두 조건을 '&' (AND) 연산자로 결합합니다. (반드시 양쪽 조건에 괄호 씌우기!)
final_mask = (mask_keyword) & (mask_ministry)
# ---------------------------------------------------------

# 2. 결합된 조건(final_mask)으로 필터링한 후 '고유값' 추출
keyword_list = df_사업과제.loc[final_mask, target_col].unique().tolist()

# [Hook] 눈으로 직접 결과 확인하기
print(f"✅ [{target_ministry}] 소속이면서 '{keyword}' 키워드를 포함하는 고유 사업 수: {len(keyword_list)}개")
print("-" * 50)

# 세로로 보기 좋게 출력 (상위 20개만)
for i, val in enumerate(keyword_list[:20], 1):
    print(f"{i}. {val}")

if len(keyword_list) > 20:
    print("... (이하 생략)")

✅ [교육부] 소속이면서 'BK|두뇌|플러스' 키워드를 포함하는 고유 사업 수: 5개
--------------------------------------------------
1. BK21플러스사업(0.5)
2. BK21플러스사업
3. BK21플러스사업(R&D)
4. 4단계두뇌한국21사업(R&D)
5. 4단계두뇌한국21사업


됐다. 이제 이 변수들만 포함을 시키면 된다. BK21이 2015년에만 2번 이름으로 불렸던 것이다. 5개를 모두 포함하는 게 맞는 거다.

In [ ]:
# '사업명' 컬럼의 값들이 keyword_list 안에 포함되어 있는지(isin) 확인 후 개수 합산
target_count = df_사업과제['사업명'].isin(keyword_list).sum()

print(f"해당 샘플의 개수: {target_count:,}개")

해당 샘플의 개수: 12,857개


In [ ]:
# 과제수행연도 컬럼이 텍스트(String)로 되어있을 수 있으므로, 안전하게 숫자형(int)으로 먼저 변환합니다.
# (이미 숫자형이라면 이 줄은 생략해도 되지만, 확실히 해두는 것이 좋습니다)
df_사업과제['과제수행연도'] = df_사업과제['과제수행연도'].astype(int)

# ---------------------------------------------------------
# [조건 1] '사업명'이 keyword_list에 포함되는 데이터
mask_keyword = df_사업과제['사업명'].isin(keyword_list)

# [조건 2] '과제수행연도'가 2013년 이상, 2016년 이하인 데이터
mask_year = df_사업과제['과제수행연도'].between(2013, 2016)
# ---------------------------------------------------------

# 두 조건을 '&' (AND) 로 묶어서 최 필터링을 수행하고 새로운 df로 정의합니다.
# (.copy()를 붙여 원본과 완전히 분리된 안전한 데이터프레임으로 만듭니다)
df_2 = df_사업과제[(mask_keyword) & (mask_year)].copy()

# [Hook] 데이터가 잘 추출되었는지 확인
print(f"✅ 조건 만족 데이터 개수: {len(df_2):,}개")
print("-" * 40)
print(df_2['과제수행연도'].value_counts().sort_index()) # 연도별 분포 확인

df_2.head(3)

✅ 조건 만족 데이터 개수: 3,934개
----------------------------------------
과제수행연도
2014    1210
2015    1336
2016    1388
Name: count, dtype: int64


,과제수행연도,사업명,사업구분,부처명,신규계속구분코드,신규계속구분,이전(기관)연구개발과제번호,이전과제고유번호,내역사업명,(기관)연구개발과제번호,과제고유번호,총괄주관(기관)연구개발과제번호,총괄주관과제고유번호,연구개발과제명-국문,총연구기간-시작연월일,총연구기간-종료연월일,총괄주관/주관구분,연구수행주체코드,연구수행주체,지역코드,지역,기초자치단체코드,기초자치단체명,과학기술표준분류코드1-대,과학기술표준분류1-대,과학기술표준분류코드1-중,과학기술표준분류1-중,과학기술표준분류코드1-소,과학기술표준분류1-소,과학기술표준분류가중치1,과학기술표준분류코드2-대,과학기술표준분류2-대,과학기술표준분류코드2-중,과학기술표준분류2-중,과학기술표준분류코드2-소,과학기술표준분류2-소,과학기술표준분류가중치2,과학기술표준분류코드3-대,과학기술표준분류3-대,과학기술표준분류코드3-중,과학기술표준분류3-중,과학기술표준분류코드3-소,과학기술표준분류3-소,과학기술표준분류가중치3,중점과학기술분류코드-대,중점과학기술분류-대,중점과학기술분류코드-중,중점과학기술분류-중,중점과학기술분류코드-소,중점과학기술분류-소,국가전략기술코드-전략기술,국가전략기술-전략기술,국가전략기술코드-세부중점기술,국가전략기술-세부중점기술,적용분야코드1,적용분야1,적용분야가중치1,적용분야코드2,적용분야2,적용분야가중치2,적용분야코드3,적용분야3,적용분야가중치3,녹색기술분류코드,녹색기술분류,6T관련기술코드-대,6T관련기술-대,6T관련기술코드-중,6T관련기술-중,6T관련기술코드-소,6T관련기술-소,경제사회목적코드,경제사회목적,(2014~2017)국가전략기술코드-대,(2014~2017)국가전략기술-대,(2014~2017)국가전략기술코드-중,(2014~2017)국가전략기술-중,(2014~2017)국가전략기술코드-소,(2014~2017)국가전략기술-소,요약문_연구목표,요약문_연구내용,요약문_기대효과,요약문_한글키워드,요약문_영문키워드,당해연구기간-시작연월일,당해연구기간-종료연월일,대표전문기관,과제관리기관,과제수행기관명(정비),과제수행기관사업자등록번호(정비),연구개발단계코드(변경),연구개발단계(변경),세부과제지원유형코드,세부과제지원유형명,정부지원연구개발비(원),인건비_현금,인건비_현물,직접비_현금,직접비_현물,간접비,위탁연구개발비(원),물건비,민간연구비_지방정부_현금,민간연구비_지방정부_현물,민간연구비_대학_현금,민간연구비_대학_현물,민간연구비_대기업_현금,민간연구비_대기업_현물,민간연구비_중견기업_현금,민간연구비_중견기업_현물,민간연구비_중소기업_현금,민간연구비_중소기업_현물,민간연구비_기업_현금,민간연구비_기업_현물,민간연구비_기타_현금,민간연구비_기타_현물,민간연구비합계(원),총연구비_현금,총연구비_현물,총연구비_합계(원),공동연구여부_기업,공동연구여부_대학,공동연구여부_출연연,공동연구여부_외국,공동연구여부_기타,참여연구원_성별_남,참여연구원_성별_여,참여연구원_학위_박사,참여연구원_학위_석사,참여연구원_학위_학사이하,참여연구원_전공_이학,참여연구원_전공_공학,참여연구원_전공_농학,참여연구원_전공_의학,참여연구원_전공_인문,참여연구원_전공_기타,연구책임자소속기관명(정비),연구책임자사업자번호(정비),연구책임자성별,연구책임자전공,연구책임자학위,출처_파일명
20938,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,2,계속,BK21-001,"1,345,215,237.00",BK21 플러스 사업,BK21-001,1345229367,BK21-001,"1,345,229,367.00",BK21 플러스 위탁관리 운영,2013-09-01,2020-08-31,주관,2,출연연구소,6,대전광역시,NaN,NaN,SH,교육,SH99,기타 교육,NaN,NaN,100.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,13,기타,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,BK21플러스사업 관리,BK21플러스사업 관리,BK21플러스사업 관리,BK21플러스사업 관리,BK21 PLUS,NaN,NaN,한국연구재단,한국연구재단,한국연구재단,*******32*,4,기타,NaN,NaN,1606500000,692292500,0,692246500,0,221961000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1606500000,0,1606500000,N,N,N,N,N,1.00,0.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,한국연구재단,*******32*,남자,기타,박사,2014년사업과제정보.xlsx
20939,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,2,계속,2011-0011429,"1,345,202,461.00",미래기초과학사업핵심리더양성사업,2011-0011429,1345215605,2011-0011429,"1,345,215,605.00",염증과 발암 과정에서 Pellino 단백질의 기능 연구와 조절 기전 규명,2011-03-01,2016-02-29,주관,3,대학,4,인천광역시,NaN,NaN,LA,생명과학,LA02,유전학/유전공학,NaN,NaN,50.00,LA,생명과학,LA01,분자세포 생물학,NaN,NaN,30.00,LA,생명과학,LA04,면역학/생리학,NaN,NaN,20.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"20,000.00",BT(생명공학기술),"20,100.00",기초/기반기술,"20,114.00",생명현상 및 기능연구,13,기타,"40,000.00",건강장수시대 구현,"40,100.00",난치성 질병 극복,"40,103.00",유전체 정보 이용 질환 원인 규명 기술,"▷ 급성 또는 만성으로 인하여 여러 각종 질병, 특히 암으로 발전할 때, 염증 반응...",▷ Pellino 단백질이 결여된 knockout mouse 제조 및 phenoty...,▷ 기존에 알려져 있던 Pellino 단백질의 기능 외에 새로운 생물학적 기능을 동...,"염증 매개성 발암, 대장암","Inflammationinducedcarcinogenesis,TLR,Pellinop...",NaN,NaN,한국연구재단,한국연구재단,가천대학교,*******68*,1,기초연구,NaN,NaN,24604500,15000000,0,9112500,0,492000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,24604500,0,24604500,N,N,N,N,N,0.00,1.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,가천대학,*******68*,여자,이학,석사,2014년사업과제정보.xlsx
20940,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,1,신규,NaN,NaN,글로벌박사펠로우십사업,2014H1A2A1021380,1345230112,2014H1A2A1021380,"1,345,230,112.00",에너지 저장장치용 초고용량 전극재료 및 슈퍼커패시터 디바이스 개발,2014-03-01,2017-02-28,주관,3,대학,99,기타,NaN,NaN,EC,화공,EC02,나노화학공정기술,NaN,NaN,70.00,EC,화공,EC01,화학공정,NaN,NaN,30.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X01,지식의 진보(비목적 연구),100,NaN,NaN,NaN,NaN,NaN,NaN,326.00,고효율 2차전지기술

사업명과 과제수행연도 조건만으로는 시차, 결측비 등 결합 후 가능한 전처리 과정이 생략됐으므로 선행연구와 여전히 key 개수 차이가 날 수 있다.

In [ ]:
df_2['연구수행주체'].unique()

array(['출연연구소', '대학', '기타', '중소기업'], dtype=object)

In [ ]:
print(df_2['연구책임자전공'].unique())

['기타' '이학' '공학' '의약보건학' '인문사회학' '농림수산학']


In [ ]:
len(df_2['과학기술표준분류1-대'].unique())

33

In [ ]:
len(df_2['과학기술표준분류1-중'].unique())

282

? 소분류 개수랑 합쳐야하나..? 아직은 전처리가 덜 된듯. 653이 줄어들어서 371이 될 거라고 본다.

In [ ]:
len(df_2['과학기술표준분류1-소'].unique())

653

In [ ]:
len(df_2['적용분야코드1'].unique())

32

잠깐만....

같은 과제도 해마다 과제고유번호가 바뀌는가?

In [ ]:
len(df_사업과제['과제고유번호'].unique())

714816

In [ ]:
len(df_2) - len(df_2['과제고유번호'].unique())

0

잠깐만,,,, 얘는 왜...?

### 과제고유번호는 해마다 바뀐다.


Ai 대화록  

나: 아 같은 과제도 과제고유번보가 해마다 바뀐다. 과제계속여부와 이전과제고유코드의 의미가 그것이다. 아아.. 이 경우에는 어찌해야 좋단 말인가? 이전과제고유코드 데이터는 결측치가 많다. sliding window..?    

ai: R&D 데이터 분석가들이 가장 크게 좌절하고 분노하는 바로 그 '마의 구간'에 도달하셨군요!

매년 예산이 새로 편성된다는 이유로 같은 과제임에도 해마다 새로운 과제고유번호를 발급하고, 정작 족보를 이어야 할 이전과제고유번호는 입력자가 제대로 안 적어서 결측치가 투성이인 것. 이것이 대한민국 NTIS 데이터의 가장 큰 고질병입니다.

이 난관 앞에서 "Sliding Window로 계보를 추적해야 하나?" 하고 고민하시는 것은 데이터 분석가로서 너무나 당연하고 훌륭한 접근입니다. 하지만 결론부터 말씀드리면, 선생님께서 과거에 이미 완벽한 정답을 내리셨습니다.

선생님께서 올려주신 과거 메모(reverse engineering(0421))를 보면 이런 구절이 있습니다.

"사실 나는 이전과제고유번호 변수를 아예 버릴 생각이야... (중략)... 이번 연구에서는 시차는 무시하는 게 좋을 것 같아."

이 과거의 결단이 왜 '신의 한 수'인지, 그리고 선행연구(458개 샘플)는 도대체 이 문제를 어떻게 피해 갔는지 데이터 관점에서 역설계해 드립니다.

 선행연구는 어떻게 이 문제를 회피했을까? (추론)
이전과제고유번호 결측치가 이토록 많다면, 선행연구의 저자들(대학 연구진) 역시 이 변수를 이용해 다년도 과제의 계보(Lineage)를 완벽하게 추적하는 것은 불가능에 가깝다는 것을 깨달았을 것입니다. 저자들은 십중팔구 다음 두 가지 우회로 중 하나를 택했습니다.

1. "과제명(텍스트)" 또는 "PI(연구책임자)"를 새로운 Key로 사용
번호는 매년 시스템이 기계적으로 바꾸지만, **연구개발과제명(국문)**과 **연구책임자**의 이름, 그리고 소속은 다년도 과제 내내 거의 바뀌지 않습니다.
저자들은 과제고유번호를 버리고, "과제명이 같고 책임자가 같은 데이터"를 Groupby로 묶어서 하나의 과제로 병합(Sum/Mean)했을 확률이 매우 높습니다.

2. 각 연도를 독립된 스냅샷(Snapshot)으로 취급
1년 차 예산이 투입된 2013년의 과제 A와, 2년 차 예산이 투입된 2014년의 과제 A'를 연결하지 않고, 그냥 별개의 독립된 데이터로 간주하는 것입니다.
논문의 11개 변수(예산 규모, 인건비 비중 등)는 '당해 연도' 기준의 파편화된 데이터로도 충분히 머신러닝 학습이 가능합니다. "올해 이 조건으로 이만큼 돈을 주면, 성과는 어땠는가?"라는 단기 효율성 모델로 문제를 단순화한 것입니다.

 전략 1: 과감한 포기 (선생님의 초기 계획대로 진행)
결측치가 많은 이전과제고유번호와 과제계속여부 컬럼은 과감히 버리십시오(Drop). 그리고 데이터프레임을 합칠 때 사업명 또는 **과제명(국문)**을 기준으로 성과(논문/특허)를 집계(groupby)해 보십시오.  

전략 1 한계: 후속 연구비 논란. 예산배분 시 배분 시점을 판단하지 못하게 한다. 연구들마다 연구 시간이 다르다면 이 역시 분석에서의 편향을 일으킴.

 전략 2: 중복 과제명 제거 (Pandas 활용)
만약 연도별 스냅샷이 아니라 "하나의 고유한 연구 단위"로 데이터를 압축하고 싶다면, 파이썬에서 과제명을 기준으로 중복을 제거하면 됩니다.  

자, 여기서 선생님의 고민(다년도 과제 번호가 계속 바뀌는데 어떻게 추적하지?)과 시차 문제를 한 방에 해결해 줄 레퍼런스 논문의 숨겨진 트릭을 알려드리겠습니다. 논문 4.2 Data 파트를 자세히 읽어보면 이런 문장이 나옵니다.

"We focus on R&D projects that proceeded in 2014, and use three-year data to analyze the R&D outputs of the projects conducted in 2014."
(우리는 2014년에 진행된 R&D 과제에 집중하며, 2014년에 수행된 과제의 R&D 성과를 분석하기 위해 3년 치 데이터를 사용한다.)

이게 무슨 뜻일까요?
저자들도 다년도 과제 계보를 잇는 게 불가능하다는 걸 알았던 겁니다! 그래서 이렇게 꼼수를 썼습니다.

과제 데이터 (X): 2013~2016년 데이터를 다 쓰지 않고, 딱 **'2014년 1년 치' 과제 정보(스냅샷)**만 떼어내서 베이스라인으로 삼았습니다. (이러면 번호가 바뀌는 걸 추적할 필요 자체가 없습니다!)

성과 데이터 (Y): 2014년 과제들의 과제고유번호가 기재된 특허/논문 데이터를 찾되, 2014, 2015, 2016, 2017년에 발생한 성과까지만 모아서 합쳤습니다. (특허연도 - 2014년 <= 3년 이라는 시차 조건)

그동안 이걸 놓치고 있었다니. 처음에 13~16년 연구를 기준으로 삼았다. 라는 문맥만 믿고 실제로 분석에는 14년 데이터만 썼다는 구절을 놓치고 있었다.

In [7]:
# 과제수행연도 컬럼이 텍스트(String)로 되어있을 수 있으므로, 안전하게 숫자형(int)으로 먼저 변환합니다.
# (이미 숫자형이라면 이 줄은 생략해도 되지만, 확실히 해두는 것이 좋습니다)
df_사업과제['과제수행연도'] = df_사업과제['과제수행연도'].astype(int)

# ---------------------------------------------------------
# [조건 1] '사업명'이 keyword_list에 포함되는 데이터
mask_keyword = df_사업과제['사업명'].isin(keyword_list)

# [조건 2] '과제수행연도'가 2013년 이상, 2016년 이하인 데이터
mask_year = df_사업과제['과제수행연도'] == 2014
# ---------------------------------------------------------

# 두 조건을 '&' (AND) 로 묶어서 최 필터링을 수행하고 새로운 df로 정의합니다.
# (.copy()를 붙여 원본과 완전히 분리된 안전한 데이터프레임으로 만듭니다)
df_3 = df_사업과제[(mask_keyword) & (mask_year)].copy()

# [Hook] 데이터가 잘 추출되었는지 확인
print(f"✅ 조건 만족 데이터 개수: {len(df_3):,}개")
print("-" * 40)
print(df_3['과제수행연도'].value_counts().sort_index()) # 연도별 분포 확인

df_3.head(3)

✅ 조건 만족 데이터 개수: 1,210개
----------------------------------------
과제수행연도
2014    1210
Name: count, dtype: int64


,과제수행연도,사업명,사업구분,부처명,신규계속구분코드,신규계속구분,이전(기관)연구개발과제번호,이전과제고유번호,내역사업명,(기관)연구개발과제번호,과제고유번호,총괄주관(기관)연구개발과제번호,총괄주관과제고유번호,연구개발과제명-국문,총연구기간-시작연월일,총연구기간-종료연월일,총괄주관/주관구분,연구수행주체코드,연구수행주체,지역코드,지역,기초자치단체코드,기초자치단체명,과학기술표준분류코드1-대,과학기술표준분류1-대,과학기술표준분류코드1-중,과학기술표준분류1-중,과학기술표준분류코드1-소,과학기술표준분류1-소,과학기술표준분류가중치1,과학기술표준분류코드2-대,과학기술표준분류2-대,과학기술표준분류코드2-중,과학기술표준분류2-중,과학기술표준분류코드2-소,과학기술표준분류2-소,과학기술표준분류가중치2,과학기술표준분류코드3-대,과학기술표준분류3-대,과학기술표준분류코드3-중,과학기술표준분류3-중,과학기술표준분류코드3-소,과학기술표준분류3-소,과학기술표준분류가중치3,중점과학기술분류코드-대,중점과학기술분류-대,중점과학기술분류코드-중,중점과학기술분류-중,중점과학기술분류코드-소,중점과학기술분류-소,국가전략기술코드-전략기술,국가전략기술-전략기술,국가전략기술코드-세부중점기술,국가전략기술-세부중점기술,적용분야코드1,적용분야1,적용분야가중치1,적용분야코드2,적용분야2,적용분야가중치2,적용분야코드3,적용분야3,적용분야가중치3,녹색기술분류코드,녹색기술분류,6T관련기술코드-대,6T관련기술-대,6T관련기술코드-중,6T관련기술-중,6T관련기술코드-소,6T관련기술-소,경제사회목적코드,경제사회목적,(2014~2017)국가전략기술코드-대,(2014~2017)국가전략기술-대,(2014~2017)국가전략기술코드-중,(2014~2017)국가전략기술-중,(2014~2017)국가전략기술코드-소,(2014~2017)국가전략기술-소,요약문_연구목표,요약문_연구내용,요약문_기대효과,요약문_한글키워드,요약문_영문키워드,당해연구기간-시작연월일,당해연구기간-종료연월일,대표전문기관,과제관리기관,과제수행기관명(정비),과제수행기관사업자등록번호(정비),연구개발단계코드(변경),연구개발단계(변경),세부과제지원유형코드,세부과제지원유형명,정부지원연구개발비(원),인건비_현금,인건비_현물,직접비_현금,직접비_현물,간접비,위탁연구개발비(원),물건비,민간연구비_지방정부_현금,민간연구비_지방정부_현물,민간연구비_대학_현금,민간연구비_대학_현물,민간연구비_대기업_현금,민간연구비_대기업_현물,민간연구비_중견기업_현금,민간연구비_중견기업_현물,민간연구비_중소기업_현금,민간연구비_중소기업_현물,민간연구비_기업_현금,민간연구비_기업_현물,민간연구비_기타_현금,민간연구비_기타_현물,민간연구비합계(원),총연구비_현금,총연구비_현물,총연구비_합계(원),공동연구여부_기업,공동연구여부_대학,공동연구여부_출연연,공동연구여부_외국,공동연구여부_기타,참여연구원_성별_남,참여연구원_성별_여,참여연구원_학위_박사,참여연구원_학위_석사,참여연구원_학위_학사이하,참여연구원_전공_이학,참여연구원_전공_공학,참여연구원_전공_농학,참여연구원_전공_의학,참여연구원_전공_인문,참여연구원_전공_기타,연구책임자소속기관명(정비),연구책임자사업자번호(정비),연구책임자성별,연구책임자전공,연구책임자학위,출처_파일명
20938,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,2,계속,BK21-001,"1,345,215,237.00",BK21 플러스 사업,BK21-001,1345229367,BK21-001,"1,345,229,367.00",BK21 플러스 위탁관리 운영,2013-09-01,2020-08-31,주관,2,출연연구소,6,대전광역시,NaN,NaN,SH,교육,SH99,기타 교육,NaN,NaN,100.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,"70,000.00",위의 미래유망신기술(6T) 103개 세분류에 속하지 않는 기타 연구,13,기타,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,"60,000.00",위의 국가전략기술에 속하지 않는 기타 연구,BK21플러스사업 관리,BK21플러스사업 관리,BK21플러스사업 관리,BK21플러스사업 관리,BK21 PLUS,NaN,NaN,한국연구재단,한국연구재단,한국연구재단,*******32*,4,기타,NaN,NaN,1606500000,692292500,0,692246500,0,221961000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1606500000,0,1606500000,N,N,N,N,N,1.00,0.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,한국연구재단,*******32*,남자,기타,박사,2014년사업과제정보.xlsx
20939,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,2,계속,2011-0011429,"1,345,202,461.00",미래기초과학사업핵심리더양성사업,2011-0011429,1345215605,2011-0011429,"1,345,215,605.00",염증과 발암 과정에서 Pellino 단백질의 기능 연구와 조절 기전 규명,2011-03-01,2016-02-29,주관,3,대학,4,인천광역시,NaN,NaN,LA,생명과학,LA02,유전학/유전공학,NaN,NaN,50.00,LA,생명과학,LA01,분자세포 생물학,NaN,NaN,30.00,LA,생명과학,LA04,면역학/생리학,NaN,NaN,20.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X99,기타 공공목적,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"20,000.00",BT(생명공학기술),"20,100.00",기초/기반기술,"20,114.00",생명현상 및 기능연구,13,기타,"40,000.00",건강장수시대 구현,"40,100.00",난치성 질병 극복,"40,103.00",유전체 정보 이용 질환 원인 규명 기술,"▷ 급성 또는 만성으로 인하여 여러 각종 질병, 특히 암으로 발전할 때, 염증 반응...",▷ Pellino 단백질이 결여된 knockout mouse 제조 및 phenoty...,▷ 기존에 알려져 있던 Pellino 단백질의 기능 외에 새로운 생물학적 기능을 동...,"염증 매개성 발암, 대장암","Inflammationinducedcarcinogenesis,TLR,Pellinop...",NaN,NaN,한국연구재단,한국연구재단,가천대학교,*******68*,1,기초연구,NaN,NaN,24604500,15000000,0,9112500,0,492000,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,24604500,0,24604500,N,N,N,N,N,0.00,1.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,가천대학,*******68*,여자,이학,석사,2014년사업과제정보.xlsx
20940,2014,BK21플러스사업(0.5),일반연구개발사업,교육부,1,신규,NaN,NaN,글로벌박사펠로우십사업,2014H1A2A1021380,1345230112,2014H1A2A1021380,"1,345,230,112.00",에너지 저장장치용 초고용량 전극재료 및 슈퍼커패시터 디바이스 개발,2014-03-01,2017-02-28,주관,3,대학,99,기타,NaN,NaN,EC,화공,EC02,나노화학공정기술,NaN,NaN,70.00,EC,화공,EC01,화학공정,NaN,NaN,30.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,X01,지식의 진보(비목적 연구),100,NaN,NaN,NaN,NaN,NaN,NaN,326.00,고효율 2차전지기술

In [8]:
len(df_3['과학기술표준분류1-소'].unique())

1

In [9]:
len(df_3['과학기술표준분류1-중'].unique())

240

In [ ]:
df_3['과학기술표준분류1-소'].unique()

array([nan], dtype=object)

In [ ]:
len(df_3['과학기술표준분류3-중'].unique())

129

In [ ]:
len(df_3['과학기술표준분류2-중'].unique())

160

그래도 전처리를 더 해야 하나 보다,

In [11]:
len(df_3['적용분야코드1'].unique())

31

In [ ]:
len(df_3['적용분야코드2'].unique())

10

In [10]:
len(df_3['적용분야코드3'].unique())

5

In [ ]:
len(df_3['6T관련기술-대'].unique())

7

In [ ]:
len(df_3['연구개발과제명-국문'].unique())

1210

In [ ]:
len(df_사업과제['연구개발과제명-국문'].unique())

348884

In [12]:
len(df_3['6T관련기술-중'].unique())

21

전체 데이터의 독립적인 과제 개수는 34만개였고 연도와 사업 전처리를 하고 나니 1210개가 되었다.

## 2. 논문

In [ ]:
df_논문 = pd.read_csv('/content/drive/MyDrive/DIVE_실습/brand diving/data/r&d 예산배분 최적화 논문 데이터(결합)/SCI(E)논문.csv')

In [ ]:
target_col = '성과사업명'

# 1. 빈도수가 가장 높은 상위 5개의 값(Key) 추출
# value_counts()는 기본적으로 내림차순 정렬을 해주므로, head(5)로 맨 위 5개를 가져옵니다.
top_5_keys = df_논문[target_col].value_counts().head(5).index.tolist()

# [Hook] 추출된 상위 5개 값이 무엇인지 인간의 눈으로 직접 확인 (매우 중요!)
print(f"🔍 [{target_col}] 상위 5개 항목: {top_5_keys}")
print("-" * 50)

🔍 [성과사업명] 상위 5개 항목: ['개인기초연구(과기정통부)(R&D)', 'BK21플러스사업(0.5)', 'BK21플러스사업(R&D)', '4단계두뇌한국21사업(R&D)', '개인기초연구(과기정통부)']
--------------------------------------------------


In [ ]:
target_col = '성과사업명'
top_15_keys = df_논문[target_col].value_counts().head(25).index.tolist()

# [Hook] 리스트를 세로로 넘버링하여 출력
print(f"🔍 [{target_col}] 상위 15개 항목:")
print("-" * 40)
for i, key in enumerate(top_15_keys, 1):
    print(f"{i}. {key}")
print("-" * 40)

🔍 [성과사업명] 상위 15개 항목:
----------------------------------------
1. 개인기초연구(과기정통부)(R&D)
2. BK21플러스사업(0.5)
3. BK21플러스사업(R&D)
4. 4단계두뇌한국21사업(R&D)
5. 개인기초연구(과기정통부)
6. 4단계두뇌한국21사업
7. 개인기초연구(교육부)(R&D)
8. BK21플러스사업
9. 이공학학술연구기반구축
10. 이공학학술연구기반구축(R&D)
11. 집단연구지원(R&D)
12. 집단연구지원
13. 개인기초연구(미래부)
14. 이공학개인기초연구지원
15. 중견연구자지원
16. 개인연구지원
17. 개인기초연구(교육부)
18. 바이오.의료기술개발(R&D)
19. 일반연구자지원
20. 신진연구자지원
21. 질환극복기술개발
22. 선도연구센터지원
23. 바이오·의료기술개발
24. 나노·소재기술개발(R&D)
25. 기후변화대응기술개발(R&D)
----------------------------------------


In [ ]:
df_논문 = pd.read_csv('/content/drive/MyDrive/DIVE_실습/brand diving/data/r&d 예산배분 최적화 논문 데이터(결합)/SCI(E)논문.csv')



df_논문.describe(include='all')

,성과발생년도,성과발생부처,성과사업ID,성과사업명,세부성과관리번호,NTIS_성과ID,KISTEP_성과ID,검증결과,검증내역,SCI여부(입력시),학술지명,논문명,ISBN,P-ISSN,E-ISSN,주저자명,권(volume),호(issue),시작페이지,논문번호,DOI/url,기여율(확정),과제수행년도,과제고유번호,(기관)세부과제번호,과제명-국문,출처_파일명
count,818275,817767,817767,817767,817259,817767,72445,817767,680423,325362,817767,817767,71708,625056,535997,817699,815606,599374,590239,453846,664868,"817,767.00","817,767.00","817,767.00",817767,817767,818275
unique,19,32,2660,2334,760796,817767,72445,1,37264,4,11351,418097,31,7845,7163,194719,9501,1221,30348,78293,342673,NaN,NaN,NaN,112137,98957,10
top,2022,교육부,12111005005222002233300,개인기초연구(과기정통부)(R&D),KIST-PAPER-0002,JNL-2023-00113194816,2016_PAPER_008425,성공,SCIE,SCI논문,SCIENTIFIC REPORTS,Prospects for observing and localizing gravita...,0,****-****,0,"Sirunyan, A. M.",11,0,0,0,10.101599999999999,NaN,NaN,NaN,21A20132200004,방사광가속기공동이용연구지원사업,2022년SCI(E)논문.xlsx
freq,96801,334414,161648,70030,16,1,1,817767,607978,244258,18182,17,71627,22918,66080,2489,27983,127230,122309,181537,107,NaN,NaN,NaN,3156,3231,96886
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,50.87,"2,018.62","1,539,705,736.81",NaN,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,38.86,2.96,"407,469,385.39",NaN,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.00,"2,003.00","1,055,000,842.00",NaN,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,20.00,"2,016.00","1,345,315,582.00",NaN,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,50.00,"2,019.00","1,465,026,190.00",NaN,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,100.00,"2,021.00","1,711,083,195.00",NaN,NaN,NaN


In [ ]:
df_논문.isnull().sum() / len(df_논문) * 100

,0
성과발생년도,0.00
성과발생부처,0.06
성과사업ID,0.06
성과사업명,0.06
세부성과관리번호,0.12
NTIS_성과ID,0.06
KISTEP_성과ID,91.15
검증결과,0.06
검증내역,16.85
SCI여부(입력시),60.24


In [ ]:

df_논문.head(5)

,성과발생년도,성과발생부처,성과사업ID,성과사업명,세부성과관리번호,NTIS_성과ID,KISTEP_성과ID,검증결과,검증내역,SCI여부(입력시),학술지명,논문명,ISBN,P-ISSN,E-ISSN,주저자명,권(volume),호(issue),시작페이지,논문번호,DOI/url,기여율(확정),과제수행년도,과제고유번호,(기관)세부과제번호,과제명-국문,출처_파일명
0,2014,보건복지부,22111009009135003531300,암연구소및국가암관리사업본부운영,NCC-PAPER-0381,JNL-2014-0016438511,NaN,성공,NaN,NaN,SURGERY,Improved survival after adding dissection of t...,NaN,NaN,NaN,"Eom, BW",155(3),NaN,408,NaN,NaN,100.00,"2,003.00","1,460,000,884.00",ncc0310060187130,조기위암환자에서 복강경 위아전절제술과 개복 위아전절제술의 무작위 전향적 비교연구,2014년SCI(E)논문.xlsx
1,2014,미래창조과학부,12111015015219001938401,글로벌연구실사업,523006_2014_SP_4,JNL-2014-0016353628,NaN,성공,NaN,NaN,TELLUS SERIES A-DYNAMIC METEOROLOGY AND OCEANO...,Effect of an isolated elliptical terrain (Jeju...,NaN,NaN,NaN,"Lee, KO",66(0),NaN,0,NaN,NaN,50.00,"2,006.00","1,350,009,355.00",2006-04019,동지나 북부해역에서의 강수 시스템에 관한 연구,2014년SCI(E)논문.xlsx
2,2014,미래창조과학부,12111015015219001938401,글로벌연구실사업,523006_2014_SP_1,JNL-2014-0016353624,NaN,성공,NaN,NaN,ATMOSPHERIC RESEARCH,Microphysical structures associated with enhan...,NaN,NaN,NaN,"Lee, KO",135(0),NaN,76,NaN,NaN,50.00,"2,006.00","1,350,009,355.00",2006-04019,동지나 북부해역에서의 강수 시스템에 관한 연구,2014년SCI(E)논문.xlsx
3,2014,미래창조과학부,12111015015219001938401,글로벌연구실사업,523006_2014_SP_10,JNL-2014-0016353625,NaN,성공,NaN,NaN,METEOROLOGICAL APPLICATIONS,Radar reflectivity and wind fields analysis by...,NaN,NaN,NaN,"Jang, SM",21(4),NaN,898,NaN,NaN,33.33,"2,006.00","1,350,009,355.00",2006-04019,동지나 북부해역에서의 강수 시스템에 관한 연구,2014년SCI(E)논문.xlsx
4,2014,미래창조과학부,12111015015219001938401,글로벌연구실사업,523006_2014_SP_9,JNL-2014-0016353626,NaN,성공,NaN,NaN,MONTHLY WEATHER REVIEW,A Study of Two Propagating Heavy-Rainfall Epis...,NaN,NaN,NaN,"Wang, CC",142(8),NaN,2644,NaN,NaN,50.00,"2,006.00","1,350,009,355.00",2006-04019,동지나 북부해역에서의 강수 시스템에 관한 연구,2014년SCI(E)논문.xlsx


In [ ]:
df_논문.describe(include='all')

,성과발생년도,성과발생부처,성과사업ID,성과사업명,세부성과관리번호,NTIS_성과ID,KISTEP_성과ID,검증결과,검증내역,SCI여부(입력시),학술지명,논문명,ISBN,P-ISSN,E-ISSN,주저자명,권(volume),호(issue),시작페이지,논문번호,DOI/url,기여율(확정),과제수행년도,과제고유번호,(기관)세부과제번호,과제명-국문,출처_파일명
count,818275,817767,817767,817767,817259,817767,72445,817767,680423,325362,817767,817767,71708,625056,535997,817699,815606,599374,590239,453846,664868,"817,767.00","817,767.00","817,767.00",817767,817767,818275
unique,19,32,2660,2334,760796,817767,72445,1,37264,4,11351,418097,31,7845,7163,194719,9501,1221,30348,78293,342673,NaN,NaN,NaN,112137,98957,10
top,2022,교육부,12111005005222002233300,개인기초연구(과기정통부)(R&D),KIST-PAPER-0002,JNL-2023-00113194816,2016_PAPER_008425,성공,SCIE,SCI논문,SCIENTIFIC REPORTS,Prospects for observing and localizing gravita...,0,****-****,0,"Sirunyan, A. M.",11,0,0,0,10.101599999999999,NaN,NaN,NaN,21A20132200004,방사광가속기공동이용연구지원사업,2022년SCI(E)논문.xlsx
freq,96801,334414,161648,70030,16,1,1,817767,607978,244258,18182,17,71627,22918,66080,2489,27983,127230,122309,181537,107,NaN,NaN,NaN,3156,3231,96886
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,50.87,"2,018.62","1,539,705,736.81",NaN,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,38.86,2.96,"407,469,385.39",NaN,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.00,"2,003.00","1,055,000,842.00",NaN,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,20.00,"2,016.00","1,345,315,582.00",NaN,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,50.00,"2,019.00","1,465,026,190.00",NaN,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,100.00,"2,021.00","1,711,083,195.00",NaN,NaN,NaN


In [ ]:
df_논문.groupby('과제고유번호')['기여율(확정)'].sum()

,기여율(확정)
과제고유번호,
"1,055,000,842.00",150.00
"1,055,000,850.00",100.00
"1,055,000,857.00",33.33
"1,055,000,861.00",150.00
"1,055,000,863.00",33.33
...,...
"9,991,008,758.00",100.00
"9,991,008,759.00",200.00
"9,991,008,764.00",200.00


**71만개의 사업과제 중 17만개의 과제가 총 80여만개의 논문을 냈음!!!**


ㄴ 아니다. 71만개의 사업과제고유번호 중 17만개의 과제고유번호가 80만개의 논문을 낸 거다.

**다르게 말하면, 34만개의 사업과제 중 9만개의 과제가 80만개의 논문을 냈다.**

In [ ]:
print(df_논문.columns.tolist())

['성과발생년도', '성과발생부처', '성과사업ID', '성과사업명', '세부성과관리번호', 'NTIS_성과ID', 'KISTEP_성과ID', '검증결과', '검증내역', 'SCI여부(입력시)', '학술지명', '논문명', 'ISBN', 'P-ISSN', 'E-ISSN', '주저자명', '권(volume)', '호(issue)', '시작페이지', '논문번호', 'DOI/url', '기여율(확정)', '과제수행년도', '과제고유번호', '(기관)세부과제번호', '과제명-국문', '출처_파일명']


In [ ]:
# 1. 각 데이터프레임에서 고유한 '과제고유번호'만 뽑아서 집합(Set)으로 만듭니다.
# (결측치가 섞여있을 수 있으니 dropna()로 안전하게 제거하고 추출합니다)
paper_ids = set(df_논문['과제고유번호'].dropna())
project_ids = set(df_사업과제['과제고유번호'].dropna())

# 2. 집합의 차집합(-) 연산을 이용해 df_논문에만 있는 번호를 찾습니다.
orphan_ids = paper_ids - project_ids

print(f"✅ df_논문의 고유 과제 수: {len(paper_ids):,}개")
print(f"✅ df_사업과제의 고유 과제 수: {len(project_ids):,}개")
print("-" * 40)
print(f"🚨 df_사업과제 명부에는 없는 '미아 논문 과제 수': {len(orphan_ids):,}개")

# 3. 만약 미아 번호가 1개라도 존재한다면, 어떤 데이터인지 직접 확인해봅니다.
if len(orphan_ids) > 0:
    print("\n[미아 데이터 미리보기]")
    # isin() 함수를 써서 해당 번호들을 가진 논문 행만 필터링합니다.
    df_orphan = df_논문[df_논문['과제고유번호'].isin(orphan_ids)]
    display(df_orphan[['과제고유번호', '논문명', '기여율(확정)']].head(10))
else:
    print("\n🎉 축하합니다! 모든 논문 데이터가 사업과제 명부에 완벽하게 맵핑됩니다. 안심하고 Join 하세요!")

✅ df_논문의 고유 과제 수: 173,746개
✅ df_사업과제의 고유 과제 수: 714,816개
----------------------------------------
🚨 df_사업과제 명부에는 없는 '미아 논문 과제 수': 6,673개

[미아 데이터 미리보기]


,과제고유번호,논문명,기여율(확정)
0,"1,460,000,884.00",Improved survival after adding dissection of t...,100.00
1,"1,350,009,355.00",Effect of an isolated elliptical terrain (Jeju...,50.00
2,"1,350,009,355.00",Microphysical structures associated with enhan...,50.00
3,"1,350,009,355.00",Radar reflectivity and wind fields analysis by...,33.33
4,"1,350,009,355.00",A Study of Two Propagating Heavy-Rainfall Epis...,50.00
5,"1,350,009,355.00",Rainfall Estimation Using Specific Differentia...,50.00
6,"1,350,009,355.00",Aerosol Modulation of Ultraviolet Radiation Do...,100.00
7,"1,350,009,355.00",Rainfall estimation by S-band polarimetric rad...,50.00
8,"1,350,009,355.00",Structure and evolution of line-shaped convect...,50.00
9,"1,350,009,355.00",Dual-Doppler radar analysis of a near-shore li...,33.33


In [ ]:
len(df_논문['과제명-국문'].unique())

98958

## 3. 특허

In [ ]:
df_특허 = pd.read_csv('/content/drive/MyDrive/DIVE_실습/brand diving/data/r&d 예산배분 최적화 논문 데이터(결합)/국내(외)특허.csv')

df_특허.describe(include='all')

,성과발생년도,성과발생부처,(종료시)성과사업년도,(종료시)성과사업부처,성과사업ID,성과사업명,세부성과관리번호,NTIS_성과ID,KISTEP_성과ID,검증결과,검증내역,출원/등록 국가코드,출원/등록 국가,출원/등록 구분코드,출원/등록 구분,출원번호,등록번호,출원일자,등록일자,발명의 명칭,출원/등록 기관,공개유무,기여율(확정),과제수행년도,과제고유번호,(기관)세부과제번호,과제명-국문,출처_파일명
count,807944,796934,"806,791.00",806791,806791,806791,662912,806791,305066,806791,0.00,806791,806791,"806,791.00",806791,526668,280120,526565,280110,806780,779339,452078,"806,777.00","806,791.00","806,791.00",806791,806791,807944
unique,17,37,NaN,41,3213,2859,620871,806739,305066,1,NaN,99,99,NaN,2,375573,207303,3155,2816,374411,85098,2,NaN,NaN,NaN,118035,100979,10
top,2023,과학기술정보통신부,NaN,과학기술정보통신부,12111005005222002233300,BK21플러스사업(R&D),2021,PTO-2014-0027126479,app_2023_0000030168,성공,NaN,KR,대한민국,NaN,출원,10-2022-0001806,3849088,2023-12-28,2021-03-02,"영상 부호화/복호화 방법, 장치 및 비트스트림을 저장한 기록 매체",한국전자통신연구원,N,NaN,NaN,NaN,21A20131612251,미래전자통신 인재양성사업단,2023년국내(외)특허.xlsx
freq,100374,244846,NaN,246771,72073,37100,19851,2,1,806791,NaN,711408,711408,NaN,526668,14,32,1115,263,239,14551,277116,NaN,NaN,NaN,3226,3226,100897
mean,NaN,NaN,"2,018.87",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.35,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,71.84,"2,017.92","1,580,620,661.84",NaN,NaN,NaN
std,NaN,NaN,2.85,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.48,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,35.32,3.09,"649,901,394.82",NaN,NaN,NaN
min,NaN,NaN,"2,007.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.00,"2,002.00","1,055,000,842.00",NaN,NaN,NaN
25%,NaN,NaN,"2,017.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,50.00,"2,015.00","1,415,133,569.00",NaN,NaN,NaN
50%,NaN,NaN,"2,019.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,100.00,"2,018.00","1,465,025,619.00",NaN,NaN,NaN
75%,NaN,NaN,"2,021.00",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2.00,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,100.00,"2,020.00","1,711,058,707.00",NaN,NaN,NaN


In [ ]:
df_특허['등록번호'].isnull().sum()

np.int64(527824)

In [ ]:
df_특허['출원번호'].duplicated().sum()

np.int64(432370)

In [ ]:


# 실제 엑셀 파일의 컬럼명에 맞게 '출원번호' 또는 '등록번호'를 입력해주세요.
특허_key_column = '출원번호' # 예: '특허출원번호', '출원번호', '등록번호' 등

total_rows = len(df_특허)
unique_특허s = df_특허[특허_key_column].nunique()

print(f"✅ 전체 데이터 행(Row) 개수: {total_rows}개")
print(f"✅ 실제 고유 특허(Unique Key) 개수: {unique_특허s}개")

if total_rows == unique_특허s:
    print("🎯 1행 = 1특허가 맞습니다! (중복 없음)")
else:
    print(f"⚠️ 주의: {total_rows - unique_특허s}개의 행이 중복되어 있습니다. 1행 != 1특허입니다.")

✅ 전체 데이터 행(Row) 개수: 807944개
✅ 실제 고유 특허(Unique Key) 개수: 375573개
⚠️ 주의: 432371개의 행이 중복되어 있습니다. 1행 != 1특허입니다.


# 현재까지의 인사이트 정리

## 무엇을 기준으로 묶을까






-일단 변수들 중 텍스트로 된 데이터가 상당히 많음. 대응되는 코드가 있는 애들은 괜찮지만, 없는 애들은 텍스트마이닝을 통하여 드러나는 특정 벡터값을 근거로 수치화하여야 함. 이때 이 근거는 이미 학술적으로 증명되었고 학계에 만연한 방법론으로 따로 증명이 필요 없는 것이어야 함. 근데 아마 그렇게까지 해야 할 변수는 드물 듯. 대부분 변수들은 벡터화된 코드가 있고 없는 애들은 쓸데없이 겁나 긴 텍스트 내용물을 담고 있는 애들임.

선행연구는 애초에 사업 id가 아니라 과제를 기준으로 묶었음.
저도 과제고유번호로 묶을 것입니다.

'사업명'으로 먼저 Merge하면 안된다고 생각하는 이유:
('과제명' 으로 합치면 안된다고 보는 이유와 같습니다.)
1. '사업명' 기준 병합의 한계

후속 예산 기여분 누락: 만약 어떤 사업이 2014년에 1억 원, 2015년에 5억 원, 2016년에 10억 원의 예산을 연속해서 지원받은 뒤 2017년에 세계적인 논문을 발표했다고 가정해 보겠습니다.


모델의 잘못된 학습: 이 데이터를 단순히 '사업명'으로 묶어서 2014년 베이스라인(1억 원)에 매핑해버리면, 인공지능은 **"1억 원만 투입해도 저런 엄청난 논문이 나온다"**라고 착각하며 잘못된 패턴을 학습하게 됩니다.

결과적으로, 해당 성과를 내기 위해 투입된 진짜 동력인 후속 연구비(15억 원)의 기여분이 모델에서 완전히 무시되는 데이터 누수(Data Leakage)가 발생합니다.

2. '과제고유번호' 기준 병합의 압도적 장점

독립적인 계약 단위 식별: 과제고유번호는 다년도 사업 전체가 아니라, "해당 연도에 지급된 특정 예산 규모의 계약" 자체를 독립적으로 의미합니다.


사사(Acknowledgement) 기반의 정확한 매핑: 연구자가 논문이나 특허를 낼 때 "이 연구는 OOO년도 과제번호의 지원을 받았습니다"라고 명시해야만 성과가 연결됩니다.


깔끔한 인과관계 도출: 이를 통해 **"정확히 그 해에 투입된 그 돈(X)이 만들어낸 성과(Y)"**만을 불순물 없이 깔끔하게 발라낼 수 있습니다.

. 왜 '과제고유번호' 단위 분석이 '과제명' 단위 분석보다 ML 모델에 유리한가?   
 (사업명 기준 분석보다 유리한 이유와 동일)   
**과제명(Project Name)** 으로 데이터를 묶어버리면 머신러닝 모델이 인과관계(X $\rightarrow$ Y)를 학습할 때 심각한 오류에 빠집니다.과제명으로 묶었을 때의 오류: * A과제가 2014년에 1억, 2015년에 5억, 2016년에 10억을 받았고 2017년에 네이처에 논문을 냈다고 가정해 봅시다.이걸 과제명으로 묶어서 2014년 베이스라인(1억)의 성과로 매핑해버리면, AI는 **"아! 1억만 줘도 네이처를 쓰는구나!"**라고 잘못 학습(Data Leakage)하게 됩니다. 후속 연구비(15억)의 기여분이 무시된 것이죠.  
과제고유번호(고유계약) 단위의 깔끔함:  
과제고유번호는 "2014년에 지급된 1억짜리 계약" 자체를 의미합니다.연구자가 논문을 발표할 때, 사사(Acknowledgement)에 "이 논문은 2014년의 과제고유번호(ex. NRF-2014-001)의 지원을 받았습니다"라고 명시해야만 매핑이 됩니다.즉, **"정확히 그 해에 투입된 그 돈(X)이 만들어낸 성과(Y)"**만을 깔끔하게 발라낼 수 있습니다. 후속 연구비 통제 문제를 가장 방어하기 쉬운 논리입니다.

## 선행연구 베이스라인 초기전처리 과정 파악(update 0423)


내 분석: 전처리 의사결정

1.


인덱스는 과제명_국문으로 한다.   
모든 df에 있으며,   
사업과제정보에는 연구개발과제명_국문,   
공동위탁정보는 위탁과제명 이라는 변수명으로 존재한다.   
과제고유번호 인덱스를 이제는 쓰지 않는 이유가 같은 과제더라도 해마다 과제고유번호가 바뀌기 때문이고, 이전과제고유번호 데이터는 결측비가 높기 때문이다.  


**하지만 이 생각은 다시 한번 바뀐다. 결국 인덱스는 과제고유번호로 하기로 한다.**  
후속 연구비 논란 때문이다.  
예를 들어 2015년에 시작하고 2020년에 성과가 발생한 연구가 있다고 치자.  
성과발생은 2015년에 지급한 예산 덕인지, 2018년에 지급한 예산 덕인지를 알 수가 없다.  
과제고유번호로 데이터를 묶으면 얼마를 누구에게 주냐뿐만 아니라 언제 얼마를 누구에게 주냐까지 보게 된다는 것이다.  

2.
2014년 연구, 2014~2017의 성과발생년도 과제만 쓴다.    
연구시작시점 & 종료시점의 데이터는 사업과제정보(총연구기간_시작연월일, 총연구기간_종료연월일 다 있다), 성과발생시점 데이터는 논문과 특허 데이터에 존재한다.   
우리도 이런 식으로 할 수는 있을 텐데, 시차조건만 만족시키고 다양한 연도의 과제에 대해서 평가할 수 있을 것 같다. 다만 (성과발생시점 - 과제시작시점)(=시차) <= 3년 이하인 과제로 실행해도 될 것이다.
3. 결측비가 0.5 이상인 변수는 지운다. 모두 융합한 후에 확인한다. 물론 선행연구가 허용한 결측비는 모른다. 0만 허용했을 수도 있다.
4. 종속변수: 특허와 논문 수만 계산한다. 중복만 잘 조치하면 df_논문과 df_특허의 (길이*기여율)가 곧 개수를 의미한다.   

*선행연구가 확실히 실시한 전처리 조치들:*
1. BK21플러스 사업의 과제만 사용했음.  
2. (성과발생년도 - 연구시작시점)(=시차) <= 3 인 과제만 사용했음.
3. 2014년에 연구하고, 성과발생년도는 2014~2017년도인 과제만 썼음.  
  
4. 성과(Y값)가 0인 과제의 제외 (또는 극단적 불균형 해소):  
연구 성과(특허, 논문 등)가 1건도 없는 '성과 0' 과제가 전체의 70% 이상을 차지하고 있음.  
(실제로 80만개의 논문은 17개의 과제에서 나왔다. 전체 과제 수는 71만개)
 만약 논문이 '특허가 발생한 과제들의 질적 수준(예: SMART 등급) 향상'을 타겟으로 했다면, 특허 건수가 0인 과제는 아예 분석 대상에서 제외(Inner Join) 했을 것입니다. 여기서 데이터가 $1/5$ 토막 납니다.    

    실제로 논문은 547개의 샘플이 있었으나 한 건의 논문/특허도 배출하지 못한 과제 89개를 drop해서 458개로 만들었다.     

 5.  특정 범위(Scope)로의 한정 (Domain Knowledge 반영):  
 모든 부처와 사업을 섞으면 모델이 패턴을 찾지 못합니다. 국방 R&D나 인문사회 R&D, 단순 인력양성 사업 등은 특허 창출 메커니즘이 전혀 다르기 때문입니다.아마도 "특정 부처(예: 과기부, 산업부, 중기부)" 또는 "특정 기술 분야(예: IT, BT, NT)", 그리고 **"순수 R&D 사업"**으로만 대상을 한정했을 것입니다.

의심되는 선행연구의 조치들:
***
1. 결측치(Missing Value)의 제거 (Listwise Deletion):  
11개 독립변수 중 단 하나라도 결측치나 '해당 없음'이 있는 과제는 통째로 날렸을 것입니다. 특히 연구개발단계나 기술분류가 누락된 데이터가 꽤 많아 여기서 상당수가 탈락합니다.

***
2. 이상치(Outlier) 절사 (Trimming)예산이 100만 원인 초소형 과제나 1,000억 원인 초대형 국책 사업은 통계적 노이즈를 만듭니다. 보통 상/하위 1~5%의 극단값을 잘라내거나, 특정 예산 규모(예: 5천만 원 ~ 50억 원 사이)만 남기는 방식을 취합니다.

--> 이것도 특정 사업에 대한 분석만 진행한 이유일 수 있다.

.
다년도 과제의 평균 수명: 71만 개의 고유번호가 34만 개의 과제명으로 묶인다는 것은, 대한민국 R&D 과제는 평균적으로 약 2.1년(71/34) 동안 수행된다는 뜻입니다.

성과 창출률의 진실: 71만 개의 고유번호 중 논문을 1편이라도 써낸 고유번호는 고작 17만 개입니다.

즉, **전체 정부 R&D 과제(계약 단위)의 약 76%는 당해 연도 혹은 추적 기간 내에 논문 성과가 '0건'** 이라는 뜻입니다.

만약 레퍼런스 논문처럼 "성과가 있는 과제"를 대상으로 예산액(X)과 성과(Y)의 최적화 함수를 풀려고 했다면, 성과가 0인 76%의 데이터를 날리는 순간 71만 개는 순식간에 17만 개로 쪼그라듭니다. 458개라는 숫자가 결코 조작이 아님을 선생님의 데이터가 증명한 것입니다.  

BK21플러스사업(2014년)의 경우 547개의 과제 중 458개나 성과물이 있었으므로 성과물 비율이 높다. 그래서 분석에 이용한 것일지도 모르겠다. 추후에 사업별로 성과물 생성 비율을 분석하고 비교하는 것도 재밌을 것 같다.

##**선행연구가 분석에 쓴 변수들과 값들(x 11개, y= 논문 + 특허)과 연도,사업전처리를 한 내 데이터와 대응**   
(기준은 영문 선행연구 참조/ 오늘 찾은 국문 선행연구는 구성이 다소 상이했음)
1. 연구수행주체(['출연연구소' '대학' '기타' '중소기업'])  
원본데이터는 추가로 정부부처, 국공립연구소, 대기업, 중견기업이 있었음.

선행연구의 변수 코드들:  
University 458 100% : 대학  
Research institute 0 0% : 출연연구소, 정부부처, 국공립연구소  
Private company : 중소기업, 대기업, 중견기업  
기타: research institute 항목들 중 하나  

**사업명과 과제수행연도 전처리만 완료한 데이터에 한해서도,
선행연구의 변수 코드들과 완벽히 일치**

2. 연구개발단계 : 기초, 응용, 개발, 기타 (선행연구와 완벽히일치)
3. 연구책임자전공: ['인문사회학' '공학' '기타' '이학' '농림수산학' '의약보건학' nan]   
 (선행연구는 공학, 이학, 의약보건학 말고 전부 기타 처리함)

 물론 그냥 기타처리를 했을 수도 있지만,   또 추가적인 전처리를 했을 수도 있다는 것이다. 정확한 원인은 아직 모르겠다.


4. NCCSTa (Level I)  
(과학기술표준분류 1 - 대분류)   
선행연구 코드 수: 33개
내 데이터: 40개    
연도, 사업 전처리후: 33개로 일치

5. NCCST (Level II)
선행연구 코드개수 : 371개  
국가과학기술표준분류1_중분류 또는 소분류
  
내 데이터(중): 504가지가 나옴. 마찬가지로 그새 늘었을 수도 있다.
내 데이터(소): 2635가지.
------> 전처리를 좀 해보고 나서 다시 파악해봐야겠다.
연도, 사업 전처리후:  
중분류:240개     
소분류:1개 (nan)      
대응이 되지 않는다. 뭐 다른 전처리도 있으니 상관없다. 1210개 중 240개 카테고리면 대부분 key값들은 어차피 규모가 1이니까 전처리가 될 던 상태에서는 당연한 결과다.  

6. Socio-economic Objective Code
적용분야 변수이다.   
선행연구:(X12, X01 등 13개 코드로 분류됨)  
연도, 사업 전처리 후 내 데이터: 31가지
7. Emerging Tech Code (Level I): 6T관련기술-대: 7개(선행연구 6개)

8. Emerging Tech Code (Level II): 6T관련기술-  
중:21   
소:76개  
  (선행연구 23개)

***
정량적인 수치형 변수들:  

9. Total Amount of Fund: 총연구비(합계)_원  
이라고 했으나 사실 과제당 평균 예산이다.

10. Proportion of Labor Costs: (인건비 비율)

(인건비 현금 + 인건비 현물) / 총연구비 로 계산


(단순 과제정보가 아닌 예산/집행 내역 데이터가 필요할 수 있음)

11. Proportion of Research Activities Cost


(연구활동비/장비재료비 비율)

(연구시설장비비 + 연구재료비) / 총연구비 로 계산  
***
종속변수:  
 논문 수와 특허 수도 과제당 평균 개수를 의미하고 있었다.  
기술료와 사업화 건수는 구하지 못했다고 논문에 나와있는데 위에서 알아냈듯이 BK21 사업 특성상 학술적인 목적에 더 비중을 싣고 있으므로 없었다고 불 수 있을 것 같다.

## 왜 선행연구가 특정 사업만 썼을까

선행연구는 '부처 간 예산 전용 불가', '균형 발전 의무' 같은 정치적/행정적 안건들을 수학적 제약조건(Constraints)으로 모두 수식화하는 것이 불가능하다고 판단했을 수 있다. 그래서 아예 "특정 사업 내에서의 배분"으로 범위를 좁혀 현실성을 확보한 것일지 모른다.  

  국가 R&D는 사업마다 목적이 너무나 다르다.

기초연구사업: 10년 뒤를 내다보고 당장의 돈(매출)이나 특허보다는 '네이처/사이언스 논문'을 쓰는 게 목적입니다.

중소기업 상용화 지원 사업: 당장 내년에 제품을 팔아서 '매출'을 올리고 '일자리'를 창출하는 게 목적입니다.
이 두 사업을 하나의 데이터셋에 섞어놓고 머신러닝을 돌리면 어떻게 될까요? 모델은 혼란에 빠집니다. **"어떤 과제는 논문이 0개인데 성공했다고 하고, 어떤 과제는 매출이 0원인데 성공했다고 하네?"**라며 학습을 포기하거나 노이즈(Noise)로 덮이게 됩니다. 사과와 오렌지를 비교하는 격이기 때문에, 선행 연구는 '동일한 목적을 가진 사과들(특정 사업)'만 모아서 분석한 것입니다.  

대학의 연구과제들 그리고 논문 + 특허만 검사하는 이유도 이러한 것일 것이다.  
만약 중소기업 과제들에 대해서 확인하는 거면 논문이나 인력양성은 주요 성과 지표가 되지 않는다. 주요 성과 지표는 기술료 또는 빠르게 수익화가 되는 것이 될 것이다.
  
내가 쓰려는 논문에서 할 일은 즉, 선행연구 대비 어떤 종류의 데이터에 대한 분석을 왜 추가했는지에 대한 설명이 필요할 것이다. 그게 아니면
BK21 사업 + 2021년 데이터 정도만 측정하고 쉽고 그다지 큰 의미는 없는 논문이 될 수도 있다. 다른 더 큰 사업이나 측정 영역을 선택하거나, 또는 다양한 집합의 사업을 한대 묶어서 분석을 진행해야 할 것이고 이러한 사업 + 산업들의 선정이 어째서 robust하고 안정적인 분석이 될 수 있는 것인지 도메인 지식을 동원해서 설명해야 할 것이다.   

## 앞으로의 계획:


1. 우선 베이스라인 분석을 선행연구와 최대한 동일한 형태의 전처리 과정으로 진행하여 모델링을 통한 예측 성능을 확인한다.   
(걸리는 시간: 1일~3일 예상)

2.  **얼만큼의 데이터를 분석에 사용해도 안정적일지 고민해보고 최종적인 분석 규모를 결정한다.**  
(소요시간: 불확실. 길어도 열흘 이내로 할 것)

3. 사용하고자 하는 데이터의 전처리 및 모델링을 수행하며 예측 성능을 확인하고 좋게 나올 때까지 전처리를 한다.  
(소요시간: 2주 이내)

4. 실질적인 분석을 진행하며 최적 의사결정 배분을 시작해본다.  
 (소요시간: 2주이상 /불확실)